File: qc-external-pg-api/src/main/java/com/eager/questioncloud/pg/implement/PGPaymentProcessor.java
package com.eager.questioncloud.pg.implement;

import com.eager.questioncloud.pg.dto.PGPayment;
import com.eager.questioncloud.pg.portone.PortoneAPI;
import com.eager.questioncloud.pg.portone.PortonePayment;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class PGPaymentProcessor {
    private final PortoneAPI portoneAPI;

    public PGPayment getPayment(String paymentId) {
        PortonePayment portonePayment = portoneAPI.getPayment(paymentId);
        return new PGPayment(portonePayment.getId(), portonePayment.getAmount().getTotal(), portonePayment.getReceiptUrl());
    }

    public void cancel(String paymentId) {
        portoneAPI.cancel(paymentId);
    }
}

File: qc-external-pg-api/src/main/java/com/eager/questioncloud/pg/implement/PGPaymentProcessor.kt
package com.eager.questioncloud.pg.implement

import com.eager.questioncloud.pg.dto.PGPayment
import com.eager.questioncloud.pg.toss.TossPaymentAPI
import org.springframework.stereotype.Component

@Component
class PGPaymentProcessor(
    private val tossPaymentAPI: TossPaymentAPI,
) {
    fun getPayment(orderId: String): PGPayment {
        val tossPayment = tossPaymentAPI.getPayment(orderId)
        return PGPayment(
            tossPayment.paymentKey,
            tossPayment.orderId,
            tossPayment.totalAmount,
            tossPayment.status,
        )
    }

    fun confirm(paymentId: String, orderId: String, amount: Int) {
        tossPaymentAPI.confirm(paymentId, orderId, amount)
    }

    fun cancel(paymentId: String, amount: Int) {
        tossPaymentAPI.cancel(paymentId, amount)
    }
}
